### DISCUSSION TITLE
should we ship SPM in BrainVisa distributions ?

### DISCUSSION THREAD

**Discussion by denisri** [brainvisa/brainvisa.github.io#106] [General]
BrainVisa provides processes that make use of external software such as SPM (but not only), which up to now need to be installed on the host system, then configured in brainvisa.
Our main pipeline, Morphologist, has an important normalization step that greatly benefits from the SPM processing when it is available. Users who don't pay attention to it just install brainvisa, and process their data directly, thus are very likely to have degraded results.
Jeff has suggested several times that we should ship SPM inside the BV distribution, already setup in brainvisa/axon, so that users directly get the best results without doing anything.

This makes a lot of sense as a user point of view.

However it requires additional packaging and setup work on our side, and we can wonder, if we provide SPM, why not FSL, Freesurfer, or all other neuroimaging software (when distribution license permissions allow us to do so, of course), so the question needs a little discussion...

**Comment by Hboni** [comment]
From a user perspective, it could be great to add SPM inside the image, as it is used inside several major pipelines as Morphologist, as you mentioned.
I think it is possible to not add other neuroimaging software as they are less used in pipelines than SPM, maybe this reason can be sufficient to justify only adding SPM.

**Comment by denisri** [comment]
I have done the additions in casa-distro to include a copy of an already installed SPM standalone in the user images. Note that there is a discussion about technical issues with spm-standalone on brainvisa/casa-distro#268.
The problem now is the size of the thing: the (recent) version of matlab MCR which allows SPM to work correctly is 8.5 GB (!) (it was less than 500 MB in spm12-standalone-7771) which is just by itself 3 times the size of the Brainvisa image...
